### PR TITLE
Include credentials in undo/redo actions

### LINKED ISSUE #593: Undo/redo actions missing credentials
If playing on a remote server, actions are validated wrt player credentials that are assumed to be inside action.payload.credentials:
https://github.com/nicolodavis/boardgame.io/blob/v0.39.3/src/master/master.ts#L188

However, in case of undo and redo, the client sends an action that consists only of type:
https://github.com/nicolodavis/boardgame.io/blob/v0.39.3/src/core/action-creators.ts#L97

Is it just me looking at the code wrong or the undo/redo actions should also include credentials?

### PR DESCRIPTION
**Pull request by tom3q**
If playing on a remote server, actions are validated wrt player
credentials that are assumed to be inside action.payload.credentials:
https://github.com/nicolodavis/boardgame.io/blob/v0.39.3/src/master/master.ts#L188

However, in case of undo and redo, the client sends an action that
consists only of type:
https://github.com/nicolodavis/boardgame.io/blob/v0.39.3/src/core/action-creators.ts#L97

Make it possible to undo/redo with a remote server and fix #593 by
adding credentials to undo/redo actions.

#### Checklist

* [v] Use a separate branch in your local repo (not `master`).
* [v] Test coverage is 100% (or you have a story for why it's ok).

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -190,6 +190,8 @@ export namespace CredentialedActionShape {
   export type AutomaticGameEvent = ReturnType<
     typeof ActionCreators.automaticGameEvent
   >;
+  export type Undo = ReturnType<typeof ActionCreators.undo>;
+  export type Redo = ReturnType<typeof ActionCreators.redo>;
   export type Any =
```
Comment (nicolodavis) [inline src/types.ts]: Can you also use the new `Undo / Redo` types in `Any` (rather than the credential-less versions being used there currently)?

**Comment by nicolodavis** [comment]
Ah, looks like this [line](https://coveralls.io/builds/29755666/source?filename=src/master/master.ts#L117) is now uncovered and doesn't have a test covering it.  Do you mind adding a test?

**Comment by tom3q** [comment]
Sure, I can take a look. Do you have an idea what could be tested to cover this line? It looks like all the remaining actions without credentials are client-only. Perhaps we should instead assert that payload and payload.credentials are in the action?

**Comment by nicolodavis** [comment]
Perhaps you can just make the function accept `CredentialedActionShape.Any` (remove `ActionShape.Any`) and then you no longer need the `if` condition?

**Comment by tom3q** [comment]
Let me check. Thanks!

**Comment by beknazar1** [comment]
Hi, is there any chance if this is ready to be merged and released any time soon?

Thanks.

**Comment by tom3q** [comment]
Updated the changes. Please take a look. :)

**Comment by maciejmatu** [comment]
That's some good stuff, waiting for it to be merged! 😄

**Pull request merged**